File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
import sys

def binary_search(arr, target):
  left, right = 0, len(arr) - 1

  while left <= right:
    mid = (left + right) // 2

    if arr[mid] == target:
      return mid
    elif arr[mid] > target:
      right = mid - 1
    else:
      left = mid + 1

  return -1

n = int(input())
arr_1 = list(map(int, sys.stdin.readline().split()))
arr_1.sort()

m = int(input())
arr_2 = list(map(int, sys.stdin.readline().split()))

for i in arr_2:
  if binary_search(arr_1, i) != -1:
    print(1, end=" ")
  else:
    print(0, end=" ")

File: 백준/Silver/10815. 숫자 카드/숫자 카드.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  
  int n, m, num;
  cin >> n;
  int *arr = new int[n];

  for (int i = 0; i < n; i++)
    cin >> arr[i];

  sort(arr, arr + n);
  
  cin >> m;

  for (int i = 0; i < m; i++) {
    cin >> num;
    if (binary_search(arr, arr + n, num))
      cout << 1 << " ";
    else
      cout << 0 << " ";
  }
}